src/storage/app-state/slices/settingsSlice.ts

```diff
@@ -59,7 +65,14 @@ const settingsSlice = createSlice({
     }
   },
   extraReducers(builder) {
-    builder.addCase(localStorageDataMigrated, () => SettingsStorage.load('general') as GeneralSettings)
+    builder
+      .addCase(localStorageDataMigrated, () => SettingsStorage.load('general') as GeneralSettings)
+      .addCase(systemLanguageMatchSucceeded, (state, action) => {
```

Comment:
Nit: To make it easier to see what the payload is, we could do:

```ts
.addCase(systemLanguageMatchSucceeded, (state, { payload: language }) => {
  state.language = language
})
```

Reply:
Absolutely!